backport new multicluster install doc

stevenctl opened the pull request:
To avoid confusion using the old `.global` approach, users still setting up multicluster with 1.7 should be using our new approach. This PR is a WIP and needs:

* the eastwest install scripts either:
  * embedded in the doc
  * dropped in favor of using ingress
  * backported to samples/ in the next 1.7.x release
* meshNetworks is still required for multi-network 

cc @nmittler 


TODO:
- [ ] Doc Updates
  - [ ] MeshNetworks config in multiNetwork docs
- [ ] Manual Verification
  - [x] Multi-Primary Single-Network
  - [ ] Multi-Primary Multi-Network
  - [ ] Primary-Remote Single-Network
    * Issue with remote-secret creation chicken-egg problem with service account. Tried to workaround by half installing, creating secret, reinstall, but ingress fails to come up. Likely requires backporting https://github.com/istio/istio/pull/28059
  - [ ] Primary-Remote Multi-Network

istio-testing commented:
@stevenctl: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
gencheck_istio.io_release-1.7 | 4e8f9835fd942655f09d3d5f7bf46d7bb8c8f109 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/8539/gencheck_istio.io_release-1.7/1329849200346664960) | `/test gencheck_istio.io_release-1.7`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@stevenctl: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
gencheck_istio.io_release-1.7 | 4e8f9835fd942655f09d3d5f7bf46d7bb8c8f109 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/8539/gencheck_istio.io_release-1.7/1329849200346664960) | `/test gencheck_istio.io_release-1.7`
lint_istio.io_release-1.7 | 4e8f9835fd942655f09d3d5f7bf46d7bb8c8f109 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/8539/lint_istio.io_release-1.7/1329849200266973184) | `/test lint_istio.io_release-1.7`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

nmittler commented:
@stevenctl is the east-west gateway in 1.7.x identical to 1.8/master? If not, maybe we should back port that first?

stevenctl commented:
https://github.com/istio/istio/pull/29090 will bring things over

istio-testing commented:
@stevenctl: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio.io_release-1.7 | 4e8f9835fd942655f09d3d5f7bf46d7bb8c8f109 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/8539/lint_istio.io_release-1.7/1329849200266973184) | `/test lint_istio.io_release-1.7`
gencheck_istio.io_release-1.7 | 1b885a5aba8058445afc935916aade143fa4110c | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/8539/gencheck_istio.io_release-1.7/1334969271905685504) | `/test gencheck_istio.io_release-1.7`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@stevenctl: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
gencheck_istio.io_release-1.7 | 1b885a5aba8058445afc935916aade143fa4110c | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/8539/gencheck_istio.io_release-1.7/1334969271905685504) | `/test gencheck_istio.io_release-1.7`
lint_istio.io_release-1.7 | 1b885a5aba8058445afc935916aade143fa4110c | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/8539/lint_istio.io_release-1.7/1334969271884713984) | `/test lint_istio.io_release-1.7`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@stevenctl: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio.io_release-1.7 | 1b885a5aba8058445afc935916aade143fa4110c | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/8539/lint_istio.io_release-1.7/1334969271884713984) | `/test lint_istio.io_release-1.7`
gencheck_istio.io_release-1.7 | e2238642a4c33f5dcea06565d8705cf9f41da5c6 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/8539/gencheck_istio.io_release-1.7/1334969988200534016) | `/test gencheck_istio.io_release-1.7`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@stevenctl: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
gencheck_istio.io_release-1.7 | e2238642a4c33f5dcea06565d8705cf9f41da5c6 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/8539/gencheck_istio.io_release-1.7/1334969988200534016) | `/test gencheck_istio.io_release-1.7`
lint_istio.io_release-1.7 | e2238642a4c33f5dcea06565d8705cf9f41da5c6 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/8539/lint_istio.io_release-1.7/1334969988179562496) | `/test lint_istio.io_release-1.7`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

stevenctl commented:
When I run the eastwest install on 1.7 I get:

```
$ISTIO/samples/multicluster/gen-eastwest-gateway.sh \                                                                  ✘ 130
    --mesh mesh1 --cluster cluster1 --network network1 | \
    istioctl --context="${CTX_CLUSTER1}" install -y -f -
✔ Ingress gateways installed
- Pruning removed resources                                                                                                                  Removed HorizontalPodAutoscaler:istio-system:istio-ingressgateway.
  Removed HorizontalPodAutoscaler:istio-system:istiod.
  Removed PodDisruptionBudget:istio-system:istiod.
  Removed PodDisruptionBudget:istio-system:istio-ingressgateway.
  Removed Deployment:istio-system:istio-ingressgateway.
  Removed Deployment:istio-system:istiod.
  Removed Service:istio-system:istio-ingressgateway.
  Removed Service:istio-system:istiod.
  Removed ConfigMap:istio-system:istio.
  Removed ConfigMap:istio-system:istio-sidecar-injector.
  Removed ServiceAccount:istio-system:istio-ingressgateway-service-account.
  Removed ServiceAccount:istio-system:istio-reader-service-account.
  Removed ServiceAccount:istio-system:istiod-service-account.
  Removed RoleBinding:istio-system:istio-ingressgateway-sds.
  Removed RoleBinding:istio-system:istiod-istio-system.
  Removed Role:istio-system:istiod-istio-system.
  Removed Role:istio-system:istio-ingressgateway-sds.
  Removed EnvoyFilter:istio-system:metadata-exchange-1.6.
  Removed EnvoyFilter:istio-system:metadata-exchange-1.7.
  Removed EnvoyFilter:istio-system:stats-filter-1.6.
  Removed EnvoyFilter:istio-system:stats-filter-1.7.
  Removed EnvoyFilter:istio-system:tcp-metadata-exchange-1.6.
  Removed EnvoyFilter:istio-system:tcp-metadata-exchange-1.7.
  Removed EnvoyFilter:istio-system:tcp-stats-filter-1.6.
  Removed EnvoyFilter:istio-system:tcp-stats-filter-1.7.
  Removed MutatingWebhookConfiguration::istio-sidecar-injector.
  Removed ValidatingWebhookConfiguration::istiod-istio-system.
  Removed ClusterRole::istio-reader-istio-system.
  Removed ClusterRole::istiod-istio-system.
  Removed ClusterRoleBinding::istio-reader-istio-system.
  Removed ClusterRoleBinding::istiod-pilot-istio-system.
✔ Installation complete
```

istio-testing commented:
@stevenctl: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>
